frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx

```diff
@@ -1,64 +1,52 @@
 import { jt, t } from "ttag";
 
-import Banner from "metabase/components/Banner";
+import { Banner } from "metabase/components/Banner";
 import ExternalLink from "metabase/core/components/ExternalLink";
 import CS from "metabase/css/core/index.css";
 import { getStoreUrl } from "metabase/selectors/settings";
 import type { TokenStatus } from "metabase-types/api";
 
 interface PaymentBannerProps {
-  isAdmin: boolean;
   tokenStatus: TokenStatus;
 }
 
-export const PaymentBanner = ({ isAdmin, tokenStatus }: PaymentBannerProps) => {
-  if (isAdmin && tokenStatus.status === "past-due") {
-    return (
-      <Banner>
-        {jt`⚠️ We couldn't process payment for your account. Please ${(
-          <ExternalLink
-            key="payment-past-due"
-            className={CS.link}
-            href={getStoreUrl()}
-          >
-            {t`review your payment settings`}
-          </ExternalLink>
-        )} to avoid service interruptions.`}
-      </Banner>
-    );
-  } else if (isAdmin && tokenStatus.status === "unpaid") {
-    return (
-      <Banner>
-        {jt`⚠️ Pro features won’t work right now due to lack of payment. ${(
-          <ExternalLink
-            key="payment-unpaid"
-            className={CS.link}
-            href={getStoreUrl()}
-          >
-            {t`Review your payment settings`}
-          </ExternalLink>
-        )} to restore Pro functionality.`}
-      </Banner>
-    );
-  } else if (isAdmin && tokenStatus.status === "invalid") {
-    return (
-      <Banner>
-        {jt`⚠️ Pro features error. ` + (tokenStatus["error-details"] || "")}
-      </Banner>
-    );
+export const PaymentBanner = ({ tokenStatus }: PaymentBannerProps) => {
+  switch (tokenStatus.status) {
```

Comment:
maybe we can use `ts-pattern`'s `match` here? we can then use `exhaustive()` to ensure that all cases of `tokenStatus.status` are covered

Reply:
agree with using `exhaustive()`, this would help ourselves in the future when we add more statuses

Reply:
I'll never get this PR over the finish line, it seems 😭 

🤣

Reply:
I don't see how `exhaustive()` helps use if the type allows for any string?
`export type TokenStatusStatus = "unpaid" | "past-due" | "invalid" | string;`

Reply:
Please see https://github.com/metabase/metabase/pull/49273/commits/e22a824ef927b90663e23d9267dc60bef4e3636a as the alternative that reflects the type more faithfully.

Reply:
> export type TokenStatusStatus = "unpaid" | "past-due" | "invalid" | string;

You're right `exhaustive()` doesn't work in this case, but this feels like a we should properly type `TokenStatusStatus` rather than having an escape hatch as `string`.